### PR TITLE
Remove IMvcRazorHost.MainClassPrefix

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak  - we spoke about refactoring this a while back. Saw this was left over when I was working on adding tests.

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs [hunk 1]
```diff
@@ -99,7 +95,7 @@ public CompilationResult Compile(RelativeFileInfo fileInfo, string compilationCo
 
             var startTimestamp = _logger.IsEnabled(LogLevel.Debug) ? Stopwatch.GetTimestamp() : 0;
 
-            var assemblyName = Path.GetRandomFileName();
+            var assemblyName = fileInfo.FileInfo.Name + "." + Path.GetRandomFileName();
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: Are all file names valid as parts of an assembly name?

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: I'll have a look.

Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: Why is the assembly name changing / getting longer here?

**Comment by dougbu** [comment]
@pranavkm what issue does this PR solve?

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: It made debugging in the Modules window a tad simpler. I'll get this tracked in a separate work item

**Comment by pranavkm** [comment]
This was some left over code that we wanted to remove the last time this file was refactored. At that point precompilation was using the property which made it tricky.

File: src/Microsoft.AspNetCore.Mvc.Razor.Host/IMvcRazorHost.cs [hunk 2]
```diff
@@ -23,11 +23,6 @@ public interface IMvcRazorHost : IDisposable
         GeneratorResults GenerateCode(string rootRelativePath, Stream inputStream);
 
         /// <summary>
-        /// Represent the prefix off the main entry class in the view.
-        /// </summary>
-        string MainClassNamePrefix { get; }
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/IMvcRazorHost.cs]: cc @Eilon

Comment (Eilon) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/IMvcRazorHost.cs]: :+1:

**Comment by pranavkm** [comment]
Updated. Moved the assembly name mangling to a separate work item - https://github.com/aspnet/Mvc/issues/4239

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs [hunk 3]
```diff
@@ -158,9 +154,8 @@ public CompilationResult Compile(RelativeFileInfo fileInfo, string compilationCo
                         assembly = LoadStream(ms, assemblySymbols: null);
                     }
 
-                    var type = assembly
-                        .GetExportedTypes()
-                        .First(t => t.Name.StartsWith(_classPrefix, StringComparison.Ordinal));
+                    Debug.Assert(assembly.GetExportedTypes().Length == 1);
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: This breaks if you have an `@functions` that defines a nested `public class`

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: Ok. I guess we can look for non-nested types.

**Comment by pranavkm** [comment]
Updated

**Comment by dougbu** [comment]
:shipit:

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request closed without merging**